spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -127,8 +138,59 @@ public DataSource cloudSqlDataSource(CloudSqlJdbcInfoProvider cloudSqlJdbcInfoPr
 		hikariConfig.setUsername(this.gcpCloudSqlProperties.getUserName());
 		hikariConfig.setPassword(this.gcpCloudSqlProperties.getPassword());
 		// Setting this is useful to disable connection initialization. Especially in unit tests.
-		hikariConfig.setInitializationFailFast(this.gcpCloudSqlProperties.getInitFailFast());
+		hikariConfig.setInitializationFailFast(this.gcpCloudSqlProperties.isInitFailFast());
 
 		return new HikariDataSource(hikariConfig);
 	}
+
+	/**
+	 * Set credentials to be used by the Google Cloud SQL socket factory.
+	 *
+	 * <p>
+	 * The only way to pass a {@link CredentialFactory} to the socket factory is by passing a class
```

Comment:
This is separate from our discussion the other day. Here I'm talking about passing a credentials factory to the socket factory, it really is the only way in which the factory receives credentials factory.

I think you're referring to passing the resource URI to `SqlCredentialsFactory` via env var and then loading the resource in `SqlCredentialsFactory` using the Spring context. That probably warrants a change below where I say "... the class can read it without any context".